### PR TITLE
added types & units and reworked some topics

### PR DESCRIPTION
**Pull request by J2thatsme**
Multisense to be changing a little bit. First round for feedback

### CODE REVIEW

**Comment by mariusgiger** [comment]
please resolve the merge conflicts

File: types/comtac/LPN_DO-2/schema.json [hunk 1]
```diff
@@ -4,31 +4,71 @@
...
+    },
+    "do_2": {
+      "title": "Channel 2",
+      "description": "Channel 2, 0 = Inactive / 1 = Active.",
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 1
+    },
     "manually": {
+      "title": "Manuall State",
```
Comment (mariusgiger) [inline types/comtac/LPN_DO-2/schema.json:71]: typo Manual

File: types/decentlab/IAM/schema.json [hunk 2]
```diff
@@ -1,52 +1,59 @@
...
     "pressure": {
+      "title": "Pressure",
+      "unit": "hpa",
       "description": "Barometric Pressure in hpa",
       "type": "number",
       "minimum": 300,
       "maximum": 1100
     },
     "pir": {
+      "title": "Pir motion",
```
Comment (mariusgiger) [inline types/decentlab/IAM/schema.json:56]: always write PIR

File: types/elsys/ers-co2/lifecycle.shema.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "irExternalTemperature": {
+      "title": "Internal External Temperature",
+      "unit": "C°",
+      "description": "irExternalTemperature",
+      "type": "number",
+      "minimum": -3276.5,
+      "maximum": 3276.5
+    },
+    "irInternalTemperature": {
+      "title": "Internal Internal Temperature",
```
Comment (mariusgiger) [inline types/elsys/ers-co2/lifecycle.shema.json:16]: adjust these titles according to the docs

File: types/elsys/ers-co2/uplink.js [hunk 4]
```diff
@@ -187,5 +187,15 @@ function DecodeElsysPayload(data) {
...
   var res = DecodeElsysPayload(hexToBytes(event.data.payload_hex));
-  emit("sample", { data: res });
+
+  if (res.vdd !== undefined) {
+    emit("sample", { topic: "lifecycle", data: { "vdd": res.vdd } });
+    delete res.vdd;
+  }
+
+  if (res.irExternalTemperature !== undefined) {
+    emit("sample", { topic: "internalTemp", data: res });
```
Comment (mariusgiger) [inline types/elsys/ers-co2/uplink.js:197]: wrong topic?

File: types/elsys/ers-desk/lifecycle.schema.json [hunk 5]
```diff
@@ -1,20 +1,32 @@
...
         "irExternalTemperature": {
+            "title": "Internal External Temperature (C°)",
+            "unit": "C°",
             "description": "irExternalTemperature",
             "type": "number",
             "minimum": -3276.5,
             "maximum": 3276.5
         },
         "irInternalTemperature": {
+            "title": "Internal Internal Temperature (C°)",
```
Comment (mariusgiger) [inline types/elsys/ers-desk/lifecycle.schema.json:16]: titles

File: types/elsys/ers-desk/lifecycle.schema.json [hunk 6]
```diff
@@ -1,20 +1,32 @@
...
+            "unit": "C°",
             "description": "irInternalTemperature",
             "type": "number",
             "minimum": -3276.5,
             "maximum": 3276.5
+        },
+        "vdd": {
```
Comment (mariusgiger) [inline types/elsys/ers-desk/lifecycle.schema.json:23]: possibly also convert this to Volt and add the voltage key

File: types/miromico/REV2_V5/meta.json [hunk 7]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "Miromico service on demand button REV2_V5",
+  "name": "Service on demand button REV2_V5",
   "version": "1.0.0",
```
Comment (mariusgiger) [inline types/miromico/REV2_V5/meta.json:3]: REV2_V5 should also be in the firmware version

File: types/miromico/REV2_V5/status.schema.json [hunk 8]
```diff
@@ -5,84 +5,100 @@
...
     "battery": {
+      "title": "Battery",
+      "unit": "V",
       "type": "number",
       "description": "Battery in volt",
       "minimum": 0,
       "maximum": 65535
     },
     "aktivButtonN": {
+      "title": "Aktive Button North",
```
Comment (mariusgiger) [inline types/miromico/REV2_V5/status.schema.json:38]: typo active

File: types/miromico/REV2_V5/status.schema.json [hunk 9]
```diff
@@ -5,84 +5,100 @@
...
       "type": "integer",
       "description": "Shows if the button north is aktive",
       "minimum": 0,
       "maximum": 1
     },
     "aktivButtonE": {
+      "title": "Aktive Button East",
```
Comment (mariusgiger) [inline types/miromico/REV2_V5/status.schema.json:45]: typo active

File: types/miromico/REV2_V5/status.schema.json [hunk 10]
```diff
@@ -5,84 +5,100 @@
...
       "type": "integer",
       "description": "Shows if the button east is aktive",
       "minimum": 0,
       "maximum": 1
     },
     "aktivButtonS": {
+      "title": "Aktive Button South",
```
Comment (mariusgiger) [inline types/miromico/REV2_V5/status.schema.json:52]: typo active

File: types/miromico/REV2_V5/status.schema.json [hunk 11]
```diff
@@ -5,84 +5,100 @@
...
       "type": "integer",
       "description": "Shows if the button south is aktive",
       "minimum": 0,
       "maximum": 1
     },
     "aktivButtonW": {
+      "title": "Aktive Button West",
```
Comment (mariusgiger) [inline types/miromico/REV2_V5/status.schema.json:59]: typo active

File: types/miromico/REV2_V5/status.schema.json [hunk 12]
```diff
@@ -5,84 +5,100 @@
...
     },
     "confirmed": {
+      "title": "Confirmed uplinks",
       "type": "integer",
       "description": "0 = confirmed uplinks, 1 = unconfirmed uplinks",
       "minimum": 0,
       "maximum": 1
     },
     "statusMessageIntervall": {
+      "title": "Status message intervall",
```
Comment (mariusgiger) [inline types/miromico/REV2_V5/status.schema.json:101]: typo interval

File: types/swisscom/multisense-ambience/ambience.schema.json [hunk 13]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (mariusgiger) [inline types/swisscom/multisense-ambience/ambience.schema.json:1]: should be ambiance

File: types/swisscom/multisense-ambience/meta.json [hunk 14]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "Multisense",
```
Comment (mariusgiger) [inline types/swisscom/multisense-ambience/meta.json:2]: write the purpose in the name

File: types/swisscom/multisense-ambience/uplink.js [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+      }
+
+    } else {
+      if (status & 0x80) {
+        emit('sample', { data: { "temperature": data.temperature, "humidity": data.humidity }, topic: "ambience" });
+        delete data.temperature;
+        delete data.humidity;
+      }
+      if (status & 0x40) {
+        emit('sample', { data: data, topic: "button_event" });
```
Comment (mariusgiger) [inline types/swisscom/multisense-ambience/uplink.js:90]: check topics

File: types/swisscom/multisense-ambience/uplink.js [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+      if (status & 0x20) {
+        emit('sample', { data: data, topic: "button_event" });
+      }
+      if (status & 0x10) {
+        emit('sample', { data: data, topic: "life_sign_event" });
+      }
+
+    } else {
+      if (status & 0x80) {
+        emit('sample', { data: { "temperature": data.temperature, "humidity": data.humidity }, topic: "ambience" });
```
Comment (mariusgiger) [inline types/swisscom/multisense-ambience/uplink.js:85]: typo ambiance

File: types/swisscom/multisense-button/meta.json [hunk 17]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "Multisense",
```
Comment (mariusgiger) [inline types/swisscom/multisense-button/meta.json:2]: purpose

File: types/swisscom/multisense-door/meta.json [hunk 18]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "Multisense",
```
Comment (mariusgiger) [inline types/swisscom/multisense-door/meta.json:2]: purpose

File: types/swisscom/multisense-vibration/meta.json [hunk 19]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "Swisscom - Multisense",
```
Comment (mariusgiger) [inline types/swisscom/multisense-vibration/meta.json:2]: purpose - remove manufacturer name

File: types/swisscom/multisense-workplace/meta.json [hunk 20]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "Multisense",
```
Comment (mariusgiger) [inline types/swisscom/multisense-workplace/meta.json:2]: purpose

**Review by mariusgiger — CHANGES_REQUESTED** [review]

File: types/comtac/LPN_DO-2/schema.json [hunk 21]
```diff
@@ -4,31 +4,71 @@
...
+      "title": "Version",
+      "description": "Device Version",
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 255
+    },
+    "snr": {
+      "title": "Serial Number",
+      "description": "Serial Number of the Device",
+      "type": "integer",
```
[the remark this answers is not in the document]
Comment (J2thatsme) [reply types/comtac/LPN_DO-2/schema.json:24]: Nice catch its Signal to Noise Ratio

**Comment by mariusgiger** [comment]
changes look good, please resolve the merge conflicts

**Pull request merged**